holoviews/core/data/dask.py

```diff
@@ -241,6 +241,30 @@ def dframe(cls, columns, dimensions):
     def nonzero(cls, dataset):
         return True
 
+    @classmethod
+    def iloc(cls, dataset, index):
+        """
+        Dask does not support iloc, therefore iloc will execute
+        the call graph and lose the laziness of the operation.
+        """
```

Comment:
I wonder if there could be optional performance warnings we could issue when laziness is lost. Not for this PR though.